flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -242,6 +242,10 @@ public Execution getCurrentExecutionAttempt() {
         return currentExecution;
     }
 
+    public List<Execution> getCurrentExecutionAttempts() {
+        return Collections.singletonList(currentExecution);
+    }
```

Comment:
```suggestion
    public Collection<Execution> getCurrentExecutions() {
        return Collections.singleton(currentExecution);
    }
```
So that it can be aligned with #20082. The benefit is that we can create an unmodifiable collection in `SpeculativeExecutionVertex` with little cost.